moveit_core/robot_trajectory/include/moveit/robot_trajectory/robot_trajectory.h

```diff
@@ -273,6 +279,56 @@ class RobotTrajectory
    */
   bool getStateAtDurationFromStart(const double request_duration, moveit::core::RobotStatePtr& output_state) const;
 
+  class Iterator : public std::iterator<std::input_iterator_tag, std::pair<moveit::core::RobotStatePtr, double>, long,
```

Comment:
Inheriting from `std::iterator` is deprecated in C++17. Instead, the iterator should just have the types that `std::iterator_traits` needs defined as members. See https://stackoverflow.com/questions/37031805/preparation-for-stditerator-being-deprecated/38103394 and https://www.fluentcpp.com/2018/05/08/std-iterator-deprecated/ for some explanation.